joblib/parallel.py

```diff
@@ -666,29 +919,69 @@ class Parallel(Logger):
         [Parallel(n_jobs=2)]: Done 6 out of 6 | elapsed:  0.0s finished
 
     '''
-    def __init__(self, n_jobs=None, backend=None, verbose=0, timeout=None,
-                 pre_dispatch='2 * n_jobs', batch_size='auto',
-                 temp_folder=None, max_nbytes='1M', mmap_mode='r',
-                 prefer=None, require=None):
-        active_backend, context_n_jobs = get_active_backend(
-            prefer=prefer, require=require, verbose=verbose)
+    def __init__(
+        self,
+        n_jobs=default_parallel_config["n_jobs"],
+        backend=None,
+        verbose=default_parallel_config["verbose"],
+        timeout=None,
+        pre_dispatch='2 * n_jobs',
+        batch_size='auto',
+        temp_folder=default_parallel_config["temp_folder"],
+        max_nbytes=default_parallel_config["max_nbytes"],
+        mmap_mode=default_parallel_config["mmap_mode"],
+        prefer=default_parallel_config["prefer"],
+        require=default_parallel_config["require"],
+    ):
+        active_backend, context_config = _get_active_backend(
+            prefer=prefer, require=require, verbose=verbose
+        )
+
         nesting_level = active_backend.nesting_level
-        if backend is None and n_jobs is None:
+
+        if backend is None and n_jobs is default_parallel_config["n_jobs"]:
             # If we are under a parallel_backend context manager, look up
             # the default number of jobs and use that instead:
-            n_jobs = context_n_jobs
-        if n_jobs is None:
+            n_jobs = context_config["n_jobs"]
+        if n_jobs is None or n_jobs is default_parallel_config["n_jobs"]:
             # No specific context override and no specific value request:
             # default to 1.
             n_jobs = 1
         self.n_jobs = n_jobs
-        self.verbose = verbose
+
+        if verbose is default_parallel_config["verbose"]:
+            self.verbose = verbose.default_value
+        else:
+            self.verbose = verbose
+
         self.timeout = timeout
         self.pre_dispatch = pre_dispatch
         self._ready_batches = queue.Queue()
         self._id = uuid4().hex
         self._reducer_callback = None
 
+        # Check if we are under a parallel_config or parallel_backend
+        # context manager and use the config from the context manager
+        # for arguments that are not explicitly set.
+        max_nbytes = _get_config_param(
+            max_nbytes, context_config, "max_nbytes"
+        )
+        temp_folder = _get_config_param(
+            temp_folder, context_config, "temp_folder"
+        )
+        mmap_mode = _get_config_param(
+            mmap_mode, context_config, "mmap_mode"
+        )
+        prefer = _get_config_param(
+            prefer, context_config, "prefer"
+        )
+        require = _get_config_param(
+            require, context_config, "require"
+        )
+        verbose = _get_config_param(
+            verbose, context_config, "verbose"
+        )
```

Comment:
I think we can make a for loop here and directly create a dictionary because this is already something that we do below. So a dict comprehension seems possible here.